internal/expr/functions/definition.go

```diff
@@ -7,6 +7,9 @@ import (
 	"github.com/genjidb/genji/internal/expr"
 )
 
+// UNLIMITED represents an unlimited number of arguments.
+const UNLIMITED = -1
```

Comment:
Full uppercase is not really idiomatic. Consider changing this to something like`variadicArity`